### PR TITLE
Various improvements for app startup

### PR DESCRIPTION
**Pull request by HaronK**
- Print application versions and build timestamp at startup
- Compare Engine and script versions
- Run tests in CI

### CODE REVIEW

File: script/Main.lua [hunk 1]
```diff
@@ -31,6 +31,10 @@ function InitSystem()
         GlobalRestrict.On()
 
         dofile('./script/Config/Version.lua')
+        if Config.gameVersion ~= ffi.string(Engine.GetVersion()) then
```
Comment (HaronK) [inline script/Main.lua:34]: Can we somehow do this without `ffi.string()`

File: engine/bin/ltr/src/main.rs [hunk 2]
```diff
@@ -41,6 +44,14 @@ extern "C" {
 }
 
 pub fn main() {
+    println!(
+        "App: {}, ver: {}, git: {}, build time: {}",
+        env!("CARGO_PKG_NAME"),
+        env!("CARGO_PKG_VERSION"),
+        GIT_VERSION,
+        BUILD_TIME
+    );
```
Comment (HaronK) [inline engine/bin/ltr/src/main.rs:53]: Will show in a log string like this:
```
App: ltr, ver: 0.1.0, git: latest-380-g03a57d95, build time: 2023-11-21 / 12:19:42 UTC
```

File: .github/workflows/pr_build.yaml [hunk 3]
```diff
@@ -54,7 +56,7 @@ jobs:
       - uses: actions/checkout@v4
       - name: Update build number
         run: |
-          sed -i 's/Config.gameVersion = \".*\"/Config.gameVersion = \"${{ needs.gen-build-version.outputs.build }}\"/' "${env:GITHUB_WORKSPACE}/script/Config/App.lua"
+          sed -i 's/Config.gameVersion = \".*\"/Config.gameVersion = \"${{ needs.gen-build-version.outputs.build }}\"/' "${env:GITHUB_WORKSPACE}/script/Config/Version.lua"
```
Comment (HaronK) [inline .github/workflows/pr_build.yaml:59]: Should this change be committed?

Comment (dgavedissian) [reply .github/workflows/pr_build.yaml:59]: No, local development should always use a sentinel value like `0.0.0`. This version should only ever be seen in packaged builds downloaded by end users.

File: engine/lib/phx/src/engine/mod.rs [hunk 4]
```diff
@@ -599,7 +599,7 @@ impl Engine {
     }
 
     pub fn get_version() -> &'static str {
-        env!("PHX_VERSION")
+        env!("CARGO_PKG_VERSION")
```
[the remark this answers is not in the document]
Comment (dgavedissian) [reply engine/lib/phx/src/engine/mod.rs:602]: I think we want to pass the version into `build.sh`, then into cargo, like `cargo build PHX_VERSION=$1` (I don't know how to actually do this properly).

File: script/Main.lua (hunk 1 above)
Comment (IllustrisJack) [reply script/Main.lua]: https://github.com/Limit-Theory-Redux/ltheory/pull/191

**Review by IllustrisJack — APPROVED** [review]

**Pull request merged**